components/nimbus/src/metrics.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{enrollment::ExperimentEnrollment, EnrollmentStatus};
+use serde_derive::{Deserialize, Serialize};
+
+pub trait MetricsHandler: Send + Sync {
+    fn record_enrollment_statuses(&self, enrollment_status_extras: Vec<EnrollmentStatusExtraDef>);
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct EnrollmentStatusExtraDef {
+    pub branch: Option<String>,
+    pub conflict_slug: Option<String>,
+    pub error_string: Option<String>,
+    pub reason: Option<String>,
+    pub slug: Option<String>,
+    pub status: Option<String>,
+}
+
+#[cfg(test)]
+impl EnrollmentStatusExtraDef {
+    pub fn branch(&self) -> &String {
```

Comment:
Nit: replace `&String` with `&str`:

```suggestion
    pub fn branch(&self) -> &str {
```